registry/storage/walk.go

```diff
@@ -38,9 +43,16 @@ func Walk(ctx context.Context, driver storageDriver.StorageDriver, from string,
 			return err
 		}
 		err = f(fileInfo)
-		skipDir := (err == ErrSkipDir)
-		if err != nil && !skipDir {
-			return err
+
+		skipDir := false
+		if err != nil {
+			if err == ErrFinishedWalk {
+				break
+			} else if err == ErrSkipDir {
+				skipDir = true
+			} else {
+				return err
```

Comment:
This condition should be sufficient to complete the walk. A `break` here will cause a `nil` return, which would work like a skip dir.